tensorflow/contrib/rnn/kernels/lstm_ops.cc

```diff
@@ -0,0 +1,599 @@
+#define EIGEN_USE_THREADS
+
+#if GOOGLE_CUDA
+#define EIGEN_USE_GPU
+#endif  // GOOGLE_CUDA
+
+#include "tensorflow/contrib/rnn/kernels/lstm_ops.h"
+
+#include <memory>
+#include <vector>
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/framework/types.h"
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/macros.h"
+
+
+namespace tensorflow {
+
+namespace {
+#if GOOGLE_CUDA
+perftools::gputools::DeviceMemory<float> AsDeviceMemory(const float* cuda_memory) {
```

Comment:
https://github.com/tensorflow/tensorflow/blob/e39d8feebb9666a331345cd8d960f5ade4652bba/tensorflow/core/common_runtime/gpu/gpu_util.h#L69 is that sufficient here?